Library/Homebrew/rubocops/lines.rb

```diff
@@ -455,6 +455,135 @@ def audit_formula(_node, _class_node, _parent_class_node, body_node)
         end
       end
 
+      # This cop makes sure that the `generate_completions_from_executable` DSL is used.
+      #
+      # @api private
+      class GenerateCompletionsDSL < FormulaCop
+        extend AutoCorrector
+
+        def audit_formula(_node, _class_node, _parent_class_node, body_node)
+          install = find_method_def(body_node, :install)
+
+          correctable_shell_completion_node(install) do |node, shell, base_name, executable, subcmd, shell_parameter|
+            # generate_completions_from_executable only applicable if shell is passed
+            next unless shell_parameter.match?(/(bash|zsh|fish)/)
+
+            base_name = base_name.delete_prefix("_").delete_suffix(".fish")
+            shell = shell.to_s.delete_suffix("_completion").to_sym
+            executable = executable.source
+            shell_parameter_stripped = shell_parameter
+                                       .delete_suffix("bash")
+                                       .delete_suffix("zsh")
+                                       .delete_suffix("fish")
+            shell_parameter_format = if shell_parameter_stripped.empty?
+              nil
+            elsif shell_parameter_stripped == "--"
+              :flag
+            elsif shell_parameter_stripped == "--shell="
+              :arg
+            else
+              "\"#{shell_parameter_stripped}\""
+            end
+
+            replacement_args = %w[]
+            replacement_args << executable
```

Comment:
No, since this is already the pathname in the form `bin/"foo"`. Calling `#inspect` would result in `"bin/\"foo\""`.

Reply:
Does it work like this? I'd have thought you'd need to capture the entire node and then call `#source` to get that to work

Reply:
It does, I suppose since it is the full Pathname node.
I figure the same procedure without `#inspect` might be working for the subcommand too, if I change the node matcher to capture the whole `str` node like this `$(str _)` instead of `(str $_)`.

Reply:
Ah, sorry, I missed the `executable = executable.source` line above. Since we don't use `executable` anywhere else I'd suggest just removing that and calling `executable.source` on this line.

Reply:
Yeah, you're totally right!